### PR TITLE
Collect xsiam items for tests even when xsoar supported

### PR DESCRIPTION
**Pull request by DinaMeylakh**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-5220)

## Description
Nightly will be handled in a different task

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by DinaMeylakh** [comment]
Working example can be found in this [commit ]([37d4526](https://github.com/demisto/content/pull/23839/commits/37d4526692e3546fb0b5482fc3ee16d4f5360de1)), this [pipeline](https://code.pan.run/xsoar/content/-/pipelines/4505270)

There you can see a changed parsing rule being installed successfully on xsiam.

File: Tests/scripts/collect_tests/constants.py
Comment (daryakoval) [inline Tests/scripts/collect_tests/constants.py:132]: Is there already `FileType.LayoutRule`? If yes please add it here, if not, @eyalpalo when it will be merged?

File: Tests/scripts/collect_tests/collect_tests.py
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:1286]:
```suggestion
    logger.info(f'collected {len(modeling_rules_to_test_str.split("\n"))} modeling rules to test:\n{modeling_rules_to_test_str}')
```

File: Tests/scripts/collect_tests/collect_tests.py
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:622]: can we use an existing method to validate marketplace compatibility here?

File: Tests/scripts/infrastructure_tests/test_collect_tests.py
Comment (dorschw) [inline Tests/scripts/infrastructure_tests/test_collect_tests.py:557]:
```suggestion
    expected_packs = ('myPack',) if file_type not in MODELING_RULE_COMPONENT_FILES | XSIAM_COMPONENT_FILES else ()
```

File: Tests/scripts/collect_tests/collect_tests.py
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:906]:
```suggestion
            if (file_type not in MODELING_RULE_COMPONENT_FILES | XSIAM_COMPONENT_FILES):
```

File: Tests/scripts/collect_tests/collect_tests.py
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:924]: can we combine it with the previous condition (modeling rule), and call a function that separates the cases? it'll be easier to follow and debug

File: Tests/scripts/collect_tests/collect_tests.py
Comment (DinaMeylakh) [reply Tests/scripts/collect_tests/collect_tests.py:1286]: https://github.com/demisto/content/actions/runs/4051727915/jobs/6970346233

File: Tests/scripts/collect_tests/collect_tests.py
Comment (DinaMeylakh) [reply Tests/scripts/collect_tests/collect_tests.py:622]: Unfortunately we can't since this is the problem we are fixing

In the __validate_marketplace_compatibity we have the following code:

``` 
case MarketplaceVersions.MarketplaceV2:
                # For XSIAM machines we collect tests that have not xsoar marketplace.
                # Tests for the packs that has only mpv2, or mpv2 and xpanse marketplaces,
                # will run on xsiam machines only.
                if (MarketplaceVersions.MarketplaceV2 not in content_item_marketplaces) or \
                        (MarketplaceVersions.XSOAR in content_item_marketplaces):
                    raise IncompatibleMarketplaceException(content_item_path, self.marketplace)
```

But if we have an xsiam component in the pack, we still want to install it on xsiam machines & this is what my code lets us do.

File: Tests/scripts/collect_tests/collect_tests.py
Comment (DinaMeylakh) [reply Tests/scripts/collect_tests/collect_tests.py:924]: Not sure that this is what you had in mind, tell me if I got it wrong.

File: Tests/scripts/collect_tests/collect_tests.py
Comment (dorschw) [reply Tests/scripts/collect_tests/collect_tests.py:622]: got it, thanks. Please add a comment explaining the difference in the code, so maintainers have an easier time changing things.

File: Tests/scripts/collect_tests/collect_tests.py
Comment (dorschw) [reply Tests/scripts/collect_tests/collect_tests.py]: I meant to have only one method for collecting xsiam content (and modeling rules are always such), and in it, have `if type is modeling rules, <then the logic of the existing flow>, else <the logic you added here>`, reducing the number of functions.

File: Packs/MicrosoftDefenderAdvancedThreatProtection/ParsingRules/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml
Comment (dorschw) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/ParsingRules/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:4]: please don't forget to remove before merging :)

File: Tests/scripts/collect_tests/collect_tests.py
Comment (DinaMeylakh) [reply Tests/scripts/collect_tests/collect_tests.py]: the problem with this is that the modeling rules is being collected through that function for nightly as well. Because our nightly accidentally works fine because of this, I don't want to be testing nightly flows for this issue. (Since on-push behaviour should be fixed asap)

But it can be handled in the following issue for fixing the nightly.
https://jira-hq.paloaltonetworks.local/browse/CIAC-5488

I'll add your comment there.

File: Packs/MicrosoftDefenderAdvancedThreatProtection/ParsingRules/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml
Comment (DinaMeylakh) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/ParsingRules/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:4]: no worries, enough release notes validations are preventing me from merging this :)

**Comment by DinaMeylakh** [comment]
checked after updates from review [here](https://code.pan.run/xsoar/content/-/jobs/21123509)

File: Tests/scripts/collect_tests/constants.py
Comment (DinaMeylakh) [reply Tests/scripts/collect_tests/constants.py:132]: seems like he didn't merge yet
https://github.com/demisto/demisto-sdk/pull/2640/files#diff-555c51fe59132215b77801a3bb499d421ebd91468c9d7806cedced34215a3048

**Review by ShahafBenYakir — APPROVED** [review]

File: Tests/scripts/collect_tests/constants.py
Comment (DinaMeylakh) [reply Tests/scripts/collect_tests/constants.py]: added it :)

**Pull request merged**
